pkg/kubelet/prober/worker.go

```diff
@@ -131,11 +134,21 @@ func newWorker(
 func (w *worker) run() {
 	probeTickerPeriod := time.Duration(w.spec.PeriodSeconds) * time.Second
 
+	// Ensure jitterSeed is between 1 and 2
+	jitterSeed := 1 + w.randomFloat - math.Floor(w.randomFloat)
+
 	// If kubelet restarted the probes could be started in rapid succession.
 	// Let the worker wait for a random portion of tickerPeriod before probing.
 	// Do it only if the kubelet has started recently.
 	if probeTickerPeriod > time.Since(w.probeManager.start) {
-		time.Sleep(time.Duration(rand.Float64() * float64(probeTickerPeriod)))
+		jitter := time.Duration(jitterSeed*float64(w.spec.PeriodSeconds)) * time.Second
+		// Override: readiness worker should start not later than startup worker,
+		// otherwise it will miss the manual trigger of the probe once started.
+		if w.probeType == readiness && w.container.StartupProbe != nil {
+			jitter = time.Duration(jitterSeed*
+				math.Min(float64(w.spec.PeriodSeconds), float64(w.container.StartupProbe.PeriodSeconds))) * time.Second
```

Comment:
Is it possible for the random value to cause this to start after the startup worker, if the startup worker's random seed is larger? Would that pose an issue?

Reply:
the seed is the same for all 3 workers per https://github.com/kubernetes/kubernetes/pull/102064/files#diff-acb3209237c09e77cb3f4c876543c4cfb7eda6b110a7ec7c52e48a920ae8c876R161